pkg/es/config/config.go

```diff
@@ -292,52 +294,60 @@ func (c *Configuration) getConfigOptions(logger *zap.Logger) ([]elastic.ClientOp
 		Timeout: c.Timeout,
 	}
 	options = append(options, elastic.SetHttpClient(httpClient))
+	options = append(options, elastic.SetBasicAuth(c.Username, c.Password))
+	transport, err := GetHTTPRoundTripper(c, logger)
+	if err != nil {
+		return nil, err
+	}
+	httpClient.Transport = transport
+	return options, nil
+}
+
+// GetHTTPRoundTripper returns configured http.RoundTripper
+func GetHTTPRoundTripper(c *Configuration, logger *zap.Logger) (http.RoundTripper, error) {
 	if c.TLS.Enabled {
 		ctlsConfig, err := c.TLS.Config()
 		if err != nil {
 			return nil, err
 		}
-		httpClient.Transport = &http.Transport{
+		return &http.Transport{
 			TLSClientConfig: ctlsConfig,
+		}, nil
+	}
+	var transport http.RoundTripper
+	httpTransport := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
```

Comment:
This is legacy code from this file, I will not refactor it in this PR. I have just moved it to a separate func so I can reuse it in other parts. The implementation indeed looks like that it could be refactored.